src/main/scala/org/constellation/p2p/PeerAPI.scala

```diff
@@ -186,25 +183,44 @@ class PeerAPI(override val ipManager: IPManager, nodeActor: ActorRef)(implicit s
                */
 
               entity(as[FinishedCheckpoint]) { fc =>
-                // TODO: Validation / etc.
-                dao.metrics.incrementMetric("peerApiRXFinishedCheckpoint")
-                onComplete(
-                  EdgeProcessor.handleFinishedCheckpoint(fc)
-                ) { result => // ^ Errors captured above
-                  val maybeResponse = result.flatten.map { _ =>
-                    val maybeData = getHostAndPortFromRemoteAddress(ip)
-                    val knownHost = maybeData.exists(
-                      i => dao.peerInfo.exists(_._2.client.hostName == i.canonicalHostName)
-                    )
-                    FinishedCheckpointResponse(!knownHost)
-                  }.toOption
-                  complete(maybeResponse)
+                optionalHeaderValueByName("ReplyTo") { replyToOpt =>
+                  val maybeData = getHostAndPortFromRemoteAddress(ip)
+                  val knownHost = maybeData.exists(
+                    i => dao.peerInfo.exists(_._2.client.hostName == i.canonicalHostName)
+                  )
+                  dao.metrics.incrementMetric("peerApiRXFinishedCheckpoint")
+                  EdgeProcessor.handleFinishedCheckpoint(fc).map { result =>
+                    replyToOpt
+                      .map(URI.create)
+                      .map { u =>
+                        makeCallback(u, FinishedCheckpointResponse(result.isSuccess))
+                      }
+                  }
+                  // TODO: wkoszycki should we allow requests from strangers?
```

Comment:
CC+ @tylerprete @buckysballs

Reply:
Also even we send back re-register info, nodes simply ignore that info. :thinking: